`from_extractor` not compatible with `RequestBodyLimitLayer`

I tried adding the `RequestBodyLimitLayer` to my application. This works for everything, except the routes that contain a layer built with `from_extractor`.

My `axum` version is `0.5.16`, `tower-http` is `0.3.4` and `rustc` is `1.63.0`.

## A minimized example

This does not compile, but it does as soon as I comment out one of the two `layer` lines.

```rust
use async_trait::async_trait;
use axum::{
    extract::{FromRequest, RequestParts},
    middleware::from_extractor,
    response::Html,
    routing::get,
    Router,
};
use std::{convert::Infallible, net::SocketAddr};
use tower_http::limit::RequestBodyLimitLayer;

#[tokio::main]
async fn main() {
    let app = Router::new()
        .layer(from_extractor::<MyExtractor>())
        .layer(RequestBodyLimitLayer::new(1))
        .route("/", get(handler));

    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
    axum::Server::bind(&addr)
        .serve(app.into_make_service())
        .await
        .unwrap();
}

async fn handler() -> Html<&'static str> {
    Html("<h1>Hello, World!</h1>")
}

struct MyExtractor;

#[async_trait]
impl<B> FromRequest<B> for MyExtractor
where
    B: Send,
{
    type Rejection = Infallible;

    async fn from_request(req: &mut RequestParts<B>) -> std::result::Result<Self, Self::Rejection> {
        todo!()
    }
}
```
## The Error Message

```
error[E0277]: the trait bound `http_body::limited::Limited<_>: Default` is not satisfied
   --> src/main.rs:15:16
    |
15  |         .layer(from_extractor::<MyExtractor>())
    |          ----- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `http_body::limited::Limited<_>`
    |          |
    |          required by a bound introduced by this call
    |
    = help: the trait `tower_service::Service<Request<ReqBody>>` is implemented for `ExtractorMiddleware<S, E>`
    = note: required for `ExtractorMiddleware<Route<http_body::limited::Limited<_>>, MyExtractor>` to implement `tower_service::Service<Request<http_body::limited::Limited<_>>>`
note: required by a bound in `Router::<B>::layer`
   --> /home/hak/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.5.16/src/routing/mod.rs:305:50
    |
305 |         L::Service: Service<Request<NewReqBody>, Response = Response<NewResBody>, Error = Infallible>
    |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Router::<B>::layer`

For more information about this error, try `rustc --explain E0277`.
```

--- a/axum/src/middleware/from_extractor.rs
+++ b/axum/src/middleware/from_extractor.rs
@@ -201,7 +201,7 @@ where
 impl<T, E, B, S> Service<Request<B>> for FromExtractor<T, E, S>
 where
     E: FromRequestParts<S> + 'static,
-    B: Default + Send + 'static,
+    B: Send + 'static,
     T: Service<Request<B>> + Clone,
     T::Response: IntoResponse,
     S: Clone + Send + Sync + 'static,
@@ -266,7 +266,6 @@ where
     E: FromRequestParts<S>,
     T: Service<Request<B>>,
     T::Response: IntoResponse,
-    B: Default,
 {
     type Output = Result<Response, T::Error>;
 
@@ -305,9 +304,10 @@ where
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::{handler::Handler, routing::get, test_helpers::*, Router};
+    use crate::{async_trait, handler::Handler, routing::get, test_helpers::*, Router};
     use axum_core::extract::FromRef;
     use http::{header, request::Parts, StatusCode};
+    use tower_http::limit::RequestBodyLimitLayer;
 
     #[tokio::test]
     async fn test_from_extractor() {
@@ -362,5 +362,30 @@ mod tests {
             .send()
             .await;
         assert_eq!(res.status(), StatusCode::OK);
+    }
+
+    // just needs to compile
+    #[allow(dead_code)]
+    fn works_with_request_body_limit() {
+        struct MyExtractor;
+
+        #[async_trait]
+        impl<S> FromRequestParts<S> for MyExtractor
+        where
+            S: Send + Sync,
+        {
+            type Rejection = std::convert::Infallible;
+
+            async fn from_request_parts(
+                _parts: &mut Parts,
+                _state: &S,
+            ) -> Result<Self, Self::Rejection> {
+                unimplemented!()
+            }
+        }
+
+        let _: Router = Router::new()
+            .layer(from_extractor::<MyExtractor>())
+            .layer(RequestBodyLimitLayer::new(1));
     }
 }

